ouroboros-consensus-test/src/Test/Util/TestMode.hs

```diff
@@ -0,0 +1,81 @@
+{-# LANGUAGE LambdaCase #-}
+-- | A @tasty@ command-line option for enabling nightly tests
+module Test.Util.TestMode (
+    IohkTestMode (..)
+  , askIohkTestMode
+  , defaultMainWithIohkTestMode
+  , iohkTestModeIngredient
+  , resetQuickCheckTests
+  ) where
+
+import           Data.Proxy (Proxy (..))
+import           Options.Applicative (metavar)
+import           Test.Tasty
+import           Test.Tasty.Ingredients
+import           Test.Tasty.Options
+import           Test.Tasty.QuickCheck
+import           Test.Util.Nightly
+
+-- | 'defaultMain' extended with 'iohkTestEnvIngredient'
+defaultMainWithIohkTestMode :: TestTree -> IO ()
+defaultMainWithIohkTestMode testTree =
+    defaultMainWithIngredients (iohkNightlyIngredient : iohkTestModeIngredient : defaultIngredients) $
+    askIohkTestMode (defaultTestEnv $ askOption iohkNightlyToTestMode)
```

Comment:
This doesn't work. The runtime option always returns `dev`, when `iohk-enable-nightly-tests` is set.